CleverPush/Source/CleverPushInstance.m

```diff
@@ -1498,7 +1498,15 @@ - (void)handleNotificationOpened:(NSDictionary*)payload isActive:(BOOL)isActive
         if (hasWebViewOpened) {
             if (notification != nil && [notification objectForKey:@"url"] != nil && [[notification objectForKey:@"url"] length] != 0 && ![[notification objectForKey:@"url"] isKindOfClass:[NSNull class]]) {
                 NSURL *url = [NSURL URLWithString:[notification objectForKey:@"url"]];
-                [CPUtils openSafari:url];
+                if ([notification objectForKey:@"autoHandleDeepLink"] != nil && ![[notification objectForKey:@"autoHandleDeepLink"] isKindOfClass:[NSNull class]] && [[notification objectForKey:@"autoHandleDeepLink"] boolValue]) {
```

Comment:
Currently you put all the logic inside `hasWebViewOpened` which is not required.
The logic needs to be put after L1496 (it does not matter if user has notification opened listener set or if hasWebViewOpened is true)